pub-matic-htb.js

```diff
@@ -79,8 +79,149 @@ function PubMaticHtb(configs) {
 
     /* Utilities
      * ---------------------------------- */
+	function __populateImprObject(returnParcels) {
+        let retArr = [],
+            impObj = {},
+            sizes = [];
+
+        returnParcels.forEach(rp => {
+            impObj = {
+                id: rp.bid_id || System.generateUniqueId(),
+                tagId: rp.xSlotRef.adUnitName,
+                bidFloor: _parseSlotParam(Constants.Keys.KADFLOOR, rp.kadfloor),
+                ext: {
+                    pmZoneId: _parseSlotParam(Constants.Keys.PMZONEID, rp.pmzoneid)
+                }
+            }
+            if (rp.hasOwnProperty(Constants.Keys.XSLOTREF) && 
+                rp.xSlotRef.hasOwnProperty(Constants.Keys.SIZES)) {
+                sizes = rp.xSlotRef.sizes[0];
+                if (sizes.length > 0) {
+                    impObj.banner = {
+                        w: sizes[0],
+                        h: sizes[1]
+                    }
+                } else {
+                    Utilities.logError("PubMatic: Error in sizes array");
+                }
+            } else {
+                Utilities.logError("PubMatic: Error in xSlotRef.sizes");
+            }
+            retArr.push(impObj);
+        });
 
-    /**
+        return retArr;
+    }	
+	
+	function _parseSlotParam(paramName, paramValue) {
+      if (!Utilities.isStr(paramValue)) {
+        paramValue && Utilities.logWarn('PubMatic: Ignoring param key: ' + paramName + ', expects string-value, found ' + typeof paramValue);
+        return Constants.UNDEFINED;
+      }
+
+      switch (paramName) {
+        case Constants.Keys.PMZONEID:
+          return paramValue.split(',').slice(0, 50).map(id => id.trim()).join();
+        case Constants.Keys.KADFLOOR:
+          return parseFloat(paramValue) || Constants.UNDEFINED;
+        case Constants.Keys.LAT:
+          return parseFloat(paramValue) || Constants.UNDEFINED;
+        case Constants.Keys.LON:
+          return parseFloat(paramValue) || Constants.UNDEFINED;
+        case Constants.Keys.YOB:
+          return parseInt(paramValue) || Constants.UNDEFINED;
+        default:
+          return paramValue;
+      }
+    }
+ 
+    function __populateImprObject(returnParcels) {
+        let retArr = [],
+            impObj = {},
+            sizes = [];
+
+        returnParcels.forEach(rp => {
+            impObj = {
+                id: rp.bid_id || System.generateUniqueId(),
+                tagId: rp.xSlotRef.adUnitName,
+                bidFloor: _parseSlotParam(Constants.Keys.KADFLOOR, rp.kadfloor),
+                ext: {
+                    pmZoneId: _parseSlotParam(Constants.Keys.PMZONEID, rp.pmzoneid)
+                }
+            }
+            if (rp.hasOwnProperty(Constants.Keys.XSLOTREF) && 
```

Comment:
This check is unnecessary as the returnedParcels should always have the same format as documented in the README.